app_architecture/ruby/ruby_on_rails.adoc

```diff
@@ -0,0 +1,323 @@
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+Ruby on Rails is a popular web framework written in https://github.com/openshift/mysql/tree/master/5.5[Ruby].
+This guide covers using Rails 4 on OpenShift.
+
+[WARNING]
+====
+We strongly advise to go through the whole tutorial to have an overview of all the steps necessary to run
+your application on the OpenShift. If you experience a problem try reading through the entire tutorial and then going back to your issue. It can also be useful to review your previous steps to ensure that all the steps were executed correctly.
+====
+
+For this guide you will need:
+
+* Basic Ruby/Rails knowledge
+* Locally installed version of Ruby 2.0.0+, Rubygems, Bundler
+* Basic Git knowledge
+* Running instance of OpenShift v3
+
+== Local Workstation Setup
+
+As first make sure that an instance of OpenShift is running and is available. For more info on
+how the get OpenShift up and running check the link:../../getting_started/administrators.html#installation-methods[installation methods]. Also make sure that your link:../../cli_reference/get_started_cli.html[`oc` CLI client is installed] and the command is accessible from your command shell, so you can use it to link:../../cli_reference/get_started_cli.html#basic-setup-and-login[log in] using the email address and password. 
+
+== Write your application
+
+If you are starting your Rails application from scratch, you need to install the Rails gem at first.
+
+----
+$ gem install rails
+Successfully installed rails-4.2.0
+1 gem installed
+----
+
+After you install the Rails gem create a new application:
+
+----
+$ rails new rails-app --database=postgresql
+----
+
+Then move into your new application directory.
+
+----
+$ cd rails-app
+----
+
+In case you already have an application, make sure the `pg` (postgresql) gem is present in your `Gemfile`. If not
+edit your `Gemfile` by adding the gem:
+
+----
+gem 'pg'
+----
+
+To generate a new `Gemfile.lock` with all your dependencies run:
+
+----
+$ bundle install
+----
+
+In addition to using the `postgresql` database with the `pg` gem, you'll also need to ensure the `config/database.yml` is using the `postgresql` adapter.
+
+Make sure the development section of your `config/database.yml` file looks like this:
+
+----
+ PostgreSQL. Versions 8.2 and up are supported.
+
+ Install the pg driver:
+   gem install pg
+ On OS X with Homebrew:
+   gem install pg -- --with-pg-config=/usr/local/bin/pg_config
+ On OS X with MacPorts:
+   gem install pg -- --with-pg-config=/opt/local/lib/postgresql84/bin/pg_config
+ On Windows:
+   gem install pg
+       Choose the win32 build.
+       Install PostgreSQL and put its /bin directory on your path.
+
+ Configure Using Gemfile
+ gem 'pg'
+
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see rails configuration guide
+  # http://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: 5
```

Comment:
nm, i see you cover that below.  you should squash all the database.yml checking/editing into a single section and single example.